src/Components/ModuleForm/ModuleForm.js

```diff
@@ -0,0 +1,310 @@
+          return newModule;
+        } else {
+          return module;
+        }
+      });
+
+      if (updatedModuleList.length === moduleList.length) {
+        console.log("No module found");
+        this.setState({
+          error: "No module found",
+          moduleList: [...moduleList],
+        });
+        event.preventDefault();
+        return;
+      }
+
+      this.setState(
+        {
+          moduleList: updatedModuleList,
+          moduleName: "",
+          grade: "",
+          userAction: "Update",
+        },
+        () => {
+          localStorage.setItem(
+            "moduleList",
+            JSON.stringify(this.state.moduleList)
+          );
+        }
+      );
+
+      // Add logic for deleting modules
+    } else if (userAction === "Delete") {
+      const newModule = { moduleName, grade };
+      const updatedModuleList = moduleList.filter(
+        (module) => module.moduleName !== newModule.moduleName
+      );
+
+      if (updatedModuleList.length === moduleList.length) {
+        console.log("No module found");
+        this.setState({
+          error: "No module found",
+          moduleList: [...moduleList],
+        });
+        event.preventDefault();
+        return;
+      }
+
+      this.setState(
+        {
+          moduleList: [...updatedModuleList],
+          moduleName: "",
+          grade: this.state.grade,
+          userAction: "Delete",
+        },
+        () => {
+          localStorage.setItem(
+            "moduleList",
+            JSON.stringify(this.state.moduleList)
+          );
+        }
+      );
+      // Add logic for deleting modules
+      console.log("Delete Module");
+    }
+
+    event.preventDefault();
+  };
+
+  handleActionChange = (event) => {
+    this.setState({
+      userAction: event.target.value,
+    });
+    console.log("Action Change");
+    event.preventDefault();
+  };
+
+  calculateGPA = () => {
+    const { moduleList } = this.state;
+    const totalModules = moduleList.length;
+    if (totalModules === 0) {
+      return 0;
+    }
+    const totalGrades = moduleList.reduce((acc, module) => {
+      switch (module.grade) {
+        case "A":
+          return acc + 5.0;
+        case "A-":
+          return acc + 4.5;
+        case "B+":
+          return acc + 4.0;
+        case "B":
+          return acc + 3.5;
+        case "B-":
+          return acc + 3.0;
+        case "C+":
+          return acc + 2.5;
+        case "C":
+          return acc + 2.0;
+        case "D":
+          return acc + 1.5;
+        default:
+          return acc;
+      }
+    }, 0);
+    return totalGrades / totalModules;
+  };
+
+  render() {
+    const { enabled, userAction, moduleName, moduleList, grade, error } =
+      this.state;
+    const gpa = this.calculateGPA();
+    return (
+      <div className="module-form-card">
+        <Card>
+          <form onSubmit={this.handleSubmit} className="module-form">
```

Comment:
Why use a form, when you assign an onClick to the button below that calls the same function? I would get rid of the form, just use the button with onClick and all good. Unless you want the user to take advantage of form functionalities, like submitting on hitting enter button etc. - was that your intention? Either way though, double defining the handleSubmit on an event handler in the same form makes not much sense